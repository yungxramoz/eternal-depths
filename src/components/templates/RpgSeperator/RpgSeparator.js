import React from 'react'
import './RpgSeparator.css'

const RpgSeparator = ({ golden = true }) => {
  const separatorClasses = ['separator']
  if (golden) separatorClasses.push('golden')
  return <hr className={separatorClasses.join(' ')} />
}

export default RpgSeparator
