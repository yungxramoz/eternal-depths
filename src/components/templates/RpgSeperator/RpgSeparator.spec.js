import React from 'react'
import TestRenderer from 'react-test-renderer'
import RpgSeparator from './RpgSeparator'

describe('RpgSeparator', () => {
  it('renders correctly', () => {
    const tree = TestRenderer.create(<RpgSeparator />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('does not apply golden class when golden prop is false', async () => {
    const testRenderer = TestRenderer.create(<RpgSeparator golden={false} />)
    const testInstance = testRenderer.root
    const hr = await testInstance.findByType('hr')
    expect(hr.props.className).not.toContain('golden')
  })
})
