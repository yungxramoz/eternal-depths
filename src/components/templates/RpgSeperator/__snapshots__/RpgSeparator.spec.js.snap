// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RpgSeparator renders correctly 1`] = `
<hr
  className="separator golden"
/>
`;
